tasks/tasks/causemos_processors_test.py

```diff
@@ -0,0 +1,415 @@
+    ]
+
+
+def gen_control_dataframe(n:int=100):
+    """
+    Returns something like:
+    random1  random2  linear1  linear2  constant
+0        51       23        0        0         5
+1        92       25        1        1         5
+2        14       88        2        2         5
+3        71       59        3        3         5
+4        60       40        4        4         5
+..      ...      ...      ...      ...       ...
+95       39        0       95       95         5
+96       84       26       96       96         5
+97       79       61       97       97         5
+98       81       76       98       98         5
+99       52        2       99       99         5
+
+[100 rows x 5 columns]
+
+    but with `n` rows.
+    """
+    numpy.random.seed(42)
+
+    # Generate linearly correlated data
+    # linear_data = np.linspace(0, 10, n)  # Linearly increasing values
+    # # Combine the data into a DataFrame
+    # data = pd.DataFrame({
+    #     "Random_1": np.random.normal(0, 1, n),
+    #     "Random_2": np.random.normal(0, 1, n),
+    #     "Linear_1": linear_data,
+    #     "Linear_2": linear_data + np.random.normal(0, 1, n)  # Add random noise to create correlation
+    # })
+
+    random1_values = numpy.random.randint(0, 101, size=n)
+    random2_values = numpy.random.randint(0, 101, size=n)
+
+    colinear1_values = numpy.arange(n)
+    colinear2_values = numpy.arange(n)
+
+    constant_values = numpy.full(n, 5)
+
+    df = pandas.DataFrame({
+        "random1": random1_values,
+        "random2": random2_values,
+        "linear1": colinear1_values,
+        "linear2": colinear2_values,
+        "constant": constant_values
+    })
+
+    return df
+
+
+
+roundToFour = partial(round, ndigits=4)
+
+
+def round_list(my_list):
+    return list(map(roundToFour, my_list))
+
+
+@pytest.mark.integration
+def test_perform_pca():
+
+    df = gen_control_dataframe(1000)
+
+    (pca_components_list, pca_components_count, pca_explained_ratios) = perform_pca(df)
+
+    assert pca_components_count == 5
+    assert list(map(lambda i: round(i, 4), pca_explained_ratios)) == [0.5004, 0.2602, 0.2393, 0.0, 0.0]
+
+    formatted = list(map(lambda i: round_list(i), pca_components_list.tolist()))
+
+    assert formatted == [
+        [0.0011, 0.0401, 0.7065, 0.7065, 0.0],
+        [0.7139, 0.6997, 0.0193, 0.0193, 0.0],
+        [0.7002, 0.7133, 0.0208, 0.0208, 0.0],
+        [0.0, 0.0, 0.7071, 0.7071, 0.0],
+        [0.0, 0.0, 0.0, 0.0, 1.0]
+    ]
+
+
+@pytest.mark.integration
+def test_pca_to_weights_v3_70_control_data():
```

Comment:
@brandomr this is me running the v3 pca-to-weight algorithm shared on slack. The pca details input is what the `perform_pca` would output for a dataset with 1000 rows and those 5 columns. This is what I call control data/set.